Mage.Tests/src/test/java/org/mage/test/cards/single/UnpredictableCycloneTest.java

```diff
@@ -0,0 +1,114 @@
+package org.mage.test.cards.single;
+
+import mage.constants.PhaseStep;
+import mage.constants.Zone;
+import org.junit.Test;
+import org.mage.test.serverside.base.CardTestPlayerBase;
+
+/**
+ * @author TheElk801
+ */
+public class UnpredictableCycloneTest extends CardTestPlayerBase {
+
+    @Test
+    public void testCyclone() {
+        // Make sure the card works normally
+
+        addCard(Zone.LIBRARY, playerA, "Goblin Piker");
+        addCard(Zone.LIBRARY, playerA, "Swamp", 10);
+        skipInitShuffling();
+        addCard(Zone.BATTLEFIELD, playerA, "Mountain");
+        addCard(Zone.BATTLEFIELD, playerA, "Unpredictable Cyclone");
+        addCard(Zone.HAND, playerA, "Desert Cerodon");
+
+        activateAbility(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Cycling");
+        setChoice(playerA, "Yes");
+
+        setStopAt(1, PhaseStep.END_TURN);
+        execute();
```

Comment:
Add `setStrictChooseMode(true);` before `execute();` to ensure that targets/choices commands aren't missing.